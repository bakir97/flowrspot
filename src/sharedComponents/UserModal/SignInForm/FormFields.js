import React from "react";
import Input from "../../Input";
import styles from "./SignInForm.module.scss";
const FormFields = ({
  first_name,
  last_name,
  date_of_birth,
  email,
  password,
  onChange,
  modalType
}) => {
  const renderInput = (value, title, name, type, full) => (
    <Input
      title={title}
      value={value}
      name={name}
      onChange={onChange}
      type={type}
      full={full}
    />
  );
  if (modalType === "login") {
    return (
      <>
        {renderInput(email, "Email Address", "email", "email", true)}
        {renderInput(password, "Password", "password", "password", true)}
      </>
    );
  }
  return (
    <>
      <div className={styles.row_inputs_container}>
        {renderInput(first_name, "First Name", "first_name", "text")}
        {renderInput(last_name, "Last Name", "last_name", "text")}
      </div>
      {renderInput(
        date_of_birth,
        "Date of Birth",
        "date_of_birth",
        "date",
        true
      )}
      {renderInput(email, "Email Address", "email", "email", true)}
      {renderInput(password, "Password", "password", "password", true)}
    </>
  );
};

export default FormFields;
