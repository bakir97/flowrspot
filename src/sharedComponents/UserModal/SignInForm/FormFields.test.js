import React from "react";
import { shallow } from "enzyme";
import FormFields from "./FormFields";
describe("FormFields", () => {
  it("should render correctly", () => {
    const component = shallow(<FormFields />);
    expect(component).toMatchSnapshot();
  });
  it("should render login inputs", () => {
    const component = shallow(<FormFields modalType="login" />);
    expect(component).toMatchSnapshot();
  });
});
