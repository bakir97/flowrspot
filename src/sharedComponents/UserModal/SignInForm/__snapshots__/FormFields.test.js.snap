// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFields should render correctly 1`] = `
<Fragment>
  <div
    className="row_inputs_container"
  >
    <Input
      name="first_name"
      title="First Name"
      type="text"
    />
    <Input
      name="last_name"
      title="Last Name"
      type="text"
    />
  </div>
  <Input
    full={true}
    name="date_of_birth"
    title="Date of Birth"
    type="date"
  />
  <Input
    full={true}
    name="email"
    title="Email Address"
    type="email"
  />
  <Input
    full={true}
    name="password"
    title="Password"
    type="password"
  />
</Fragment>
`;

exports[`FormFields should render login inputs 1`] = `
<Fragment>
  <Input
    full={true}
    name="email"
    title="Email Address"
    type="email"
  />
  <Input
    full={true}
    name="password"
    title="Password"
    type="password"
  />
</Fragment>
`;
